### PR TITLE
Update CHEATSHEET.md

### PR DESCRIPTION
**Pull request by GQH1993**
## Description

Added a way to pre-check whether the ConfigMap resource exists and avoid temporary objects returned when the get() function is called directly.
## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by rohanKanojia** [comment]
I haven't tried `isReady` on a ConfigMap myself so not sure about this. Does it work when you use `isReady` with ConfigMap? I think it should throw error since `ConfigMap` is not Readiable resource:
https://github.com/fabric8io/kubernetes-client/blob/ec1e20eca8996db6a6ed0b208c8e04da1b7b7b8b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/readiness/Readiness.java#L46-L57

**Comment by manusa** [comment]
> ...I haven't tried `isReady`...

ConfigMaps should be "ready" as soon as they exist, don't think there are any status or conditions.

File: doc/CHEATSHEET.md [hunk 1]
```diff
@@ -688,6 +688,10 @@ ConfigMap configMap = client.configMaps().load(new FileInputStream("configmap1.y
 ```
 ConfigMap configMap = client.configMaps().inNamespace("default").withName("configmap1").get();
 ```
+- Check `ConfigMap` from API server:
+```
+Boolean ready = client.configMaps().inNamespace("default").withName("configmap1").isReady();
```
Comment (rohanKanojia) [inline doc/CHEATSHEET.md:693]: @GQH1993 : Hi, When I try to run this code like this:
```java
try (KubernetesClient client = new DefaultKubernetesClient()) {
    client.configMaps().inNamespace("default").withName("cm1").isReady();
}
```

I'm getting this error:
```
/usr/java/jdk-14.0.1/bin/java -javaagent:/opt/idea-IC-202.6397.94/lib/idea_rt.jar=33837:/opt/idea-IC-202.6397.94/bin -Dfile.encoding=UTF-8 -classpath /home/rohaan/work/repos/timepassproject/target/classes:/home/rohaan/.m2/repository/io/fabric8/kubernetes-client/4.12-SNAPSHOT/kubernetes-client-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-core/4.12-SNAPSHOT/kubernetes-model-core-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-common/4.12-SNAPSHOT/kubernetes-model-common-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/javax/annotation/javax.annotation-api/1.3.2/javax.annotation-api-1.3.2.jar:/home/rohaan/.m2/repository/javax/xml/bind/jaxb-api/2.3.0/jaxb-api-2.3.0.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-rbac/4.12-SNAPSHOT/kubernetes-model-rbac-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-admissionregistration/4.12-SNAPSHOT/kubernetes-model-admissionregistration-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-apps/4.12-SNAPSHOT/kubernetes-model-apps-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-autoscaling/4.12-SNAPSHOT/kubernetes-model-autoscaling-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-apiextensions/4.12-SNAPSHOT/kubernetes-model-apiextensions-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-batch/4.12-SNAPSHOT/kubernetes-model-batch-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-certificates/4.12-SNAPSHOT/kubernetes-model-certificates-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-coordination/4.12-SNAPSHOT/kubernetes-model-coordination-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-discovery/4.12-SNAPSHOT/kubernetes-model-discovery-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-events/4.12-SNAPSHOT/kubernetes-model-events-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-extensions/4.12-SNAPSHOT/kubernetes-model-extensions-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-networking/4.12-SNAPSHOT/kubernetes-model-networking-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-metrics/4.12-SNAPSHOT/kubernetes-model-metrics-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-policy/4.12-SNAPSHOT/kubernetes-model-policy-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-scheduling/4.12-SNAPSHOT/kubernetes-model-scheduling-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-settings/4.12-SNAPSHOT/kubernetes-model-settings-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-storageclass/4.12-SNAPSHOT/kubernetes-model-storageclass-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/kubernetes-model-node/4.12-SNAPSHOT/kubernetes-model-node-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/com/squareup/okhttp3/okhttp/3.12.12/okhttp-3.12.12.jar:/home/rohaan/.m2/repository/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar:/home/rohaan/.m2/repository/com/squareup/okhttp3/logging-interceptor/3.12.12/logging-interceptor-3.12.12.jar:/home/rohaan/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.11.2/jackson-dataformat-yaml-2.11.2.jar:/home/rohaan/.m2/repository/org/yaml/snakeyaml/1.26/snakeyaml-1.26.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.11.2/jackson-datatype-jsr310-2.11.2.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.11.2/jackson-annotations-2.11.2.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.11.2/jackson-databind-2.11.2.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.11.2/jackson-core-2.11.2.jar:/home/rohaan/.m2/repository/io/fabric8/zjsonpatch/0.3.0/zjsonpatch-0.3.0.jar:/home/rohaan/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2.jar:/home/rohaan/.m2/repository/dk/brics/automaton/automaton/1.11-8/automaton-1.11-8.jar:/home/rohaan/.m2/repository/io/fabric8/openshift-client/4.12-SNAPSHOT/openshift-client-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/openshift-model/4.12-SNAPSHOT/openshift-model-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/openshift-model-operator/4.12-SNAPSHOT/openshift-model-operator-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/openshift-model-operatorhub/4.12-SNAPSHOT/openshift-model-operatorhub-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/openshift-model-monitoring/4.12-SNAPSHOT/openshift-model-monitoring-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/openshift-model-console/4.12-SNAPSHOT/openshift-model-console-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/tekton-client/4.12-SNAPSHOT/tekton-client-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/tekton-model-v1alpha1/4.12-SNAPSHOT/tekton-model-v1alpha1-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/tekton-model-v1beta1/4.12-SNAPSHOT/tekton-model-v1beta1-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/tekton-model-triggers/4.12-SNAPSHOT/tekton-model-triggers-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/2.11.2/jackson-module-jaxb-annotations-2.11.2.jar:/home/rohaan/.m2/repository/jakarta/xml/bind/jakarta.xml.bind-api/2.3.2/jakarta.xml.bind-api-2.3.2.jar:/home/rohaan/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.1/jakarta.activation-api-1.2.1.jar:/home/rohaan/.m2/repository/commons-codec/commons-codec/1.15/commons-codec-1.15.jar:/home/rohaan/.m2/repository/org/apache/commons/commons-compress/1.20/commons-compress-1.20.jar:/home/rohaan/.m2/repository/org/slf4j/slf4j-simple/1.7.30/slf4j-simple-1.7.30.jar:/home/rohaan/.m2/repository/org/json/json/20190722/json-20190722.jar:/home/rohaan/.m2/repository/io/fabric8/knative-client/4.12-SNAPSHOT/knative-client-4.12-SNAPSHOT.jar:/home/rohaan/.m2/repository/io/fabric8/knative-model/4.12-SNAPSHOT/knative-model-4.12-SNAPSHOT.jar io.fabric8.ConfigMapReady
Exception in thread "main" java.lang.IllegalArgumentException: Item needs to be one of [Node, Deployment, ReplicaSet, StatefulSet, Pod, ReplicationController], but was: [ConfigMap]
	at io.fabric8.kubernetes.client.internal.readiness.Readiness.isReady(Readiness.java:62)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.isReady(BaseOperation.java:1102)
	at io.fabric8.ConfigMapReady.main(ConfigMapReady.java:9)
```

**Comment by rohanKanojia** [comment]
@GQH1993 : polite ping, Could you please elaborate on your use case a bit? I tried out the code in your PR but it doesn't seem to be working for me. Maybe I'm missing something?

File: doc/CHEATSHEET.md (hunk 1 above)
Comment (GQH1993) [reply doc/CHEATSHEET.md:693]: You are right. This was an oversight on my part。.
I think I can provide validation methods for ConfigMap resources. Currently, I find that null can only be determined by get().

**Comment by rohanKanojia** [comment]
A polite ping, still working on this patch?

**Pull request closed without merging**